kgo.js

```diff
@@ -9,27 +9,33 @@ const R = require('ramda');
 const S = require('sanctuary');
 
 
-const readFile = (filename, callback) => fs.readFile(filename, {encoding: 'utf8'}, callback);
-const complete = (error, result) => {
-  if(error == null){
-    process.stdout.write(result);
-    process.exit(0);
-  }else{
-    process.stderr.write(String(error) + '\n');
-    process.exit(1);
-  }
-};
+// join :: String -> String -> String
+const join = R.curryN(2, path.join);
+
+// data Text = Buffer | String
+// readFile :: String -> String -> ((Error?, Text?) -> Unit) -> Unit
+const readFile = R.curry((encoding, filename, callback) => {
+  fs.readFile(filename, {encoding: encoding}, callback);
+});
 
 
 const main = () => {
-  const relative = filename => path.join(process.argv[2], filename);
+  const dir = process.argv[2];
 
   kgo
-  ('index', R.partial(readFile, [relative('index.txt')]))
-  ('filePaths', ['index'], kgo.sync(S.compose(R.map(relative), S.lines)))
-  ('files', ['filePaths'], R.partial(foreign.parallel, [readFile]))
+  ('index', readFile('utf8', join(dir, 'index.txt')))
+  ('filePaths', ['index'], kgo.sync(S.compose(R.map(join(dir)), S.lines)))
+  ('files', ['filePaths'], R.partial(foreign.parallel, [readFile('utf8')]))
   ('concated', ['files'], kgo.sync(R.join('')))
-  (['*', 'concated'], complete);
+  (['*', 'concated'], (err, data) => {
```

Comment:
Because kgo is used with CPS functions a lot, you would normally pass a callback as a task like:

```
(['*','result', callback)
```

But, since this example handles the result and error directly in the main function, it makes more sense to split them into their own tasks.

I'd personally prefer the split approach for this example.